src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -956,6 +1070,68 @@ public DocumentId AddTextContainer(SourceTextContainer textContainer, string ful
                 return documentId;
             }
 
+            public void AddDynamicFile(IDynamicFileInfoProvider fileInfoProvider, string filePath, DynamicFileInfo fileInfo, ImmutableArray<string> folders)
+            {
+                var documentInfo = CreateDocumentInfoFromFileInfo(fileInfo, folders.IsDefault ? null : (IEnumerable<string>)folders);
+                var documentId = documentInfo.Id;
+
+                lock (_project._gate)
+                {
+                    if (_documentPathsToDocumentIds.ContainsKey(filePath))
```

Comment:
existing code made sure filepath given from davkean never leaked to workspace and _documentPathsToDOcumentIds points to the filepath from davkean, but now I changed it so that, _documentPathsToDocumentIds points to filepath we give to workspace. still filepath from davkean never leaked to workspace.